File: app/src/main/java/com/github/sewerina/meter_readings/database/ReadingEntity.java
package com.github.sewerina.meter_readings.database;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverters;

import java.io.Serializable;
import java.util.Date;

@Entity(tableName = "reading",
        foreignKeys = @ForeignKey(entity = HomeEntity.class,
                parentColumns = "id",
                childColumns = "homeId"), indices = {@Index("homeId")})
public class ReadingEntity implements Serializable {
    @ColumnInfo(name = "homeId")
    public int homeId;

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    public int id;

    @ColumnInfo(name = "date")
    @TypeConverters({DateConverter.class})
    public Date date;

    @ColumnInfo(name = "coldWater")
    public int coldWater;

    @ColumnInfo(name = "hotWater")
    public int hotWater;

    @ColumnInfo(name = "drainWater")
    public int drainWater;

    @ColumnInfo(name = "electricity")
    public int electricity;

    @ColumnInfo(name = "gas")
    public int gas;

    public ReadingEntity() {

    }

}

File: app/src/main/java/com/github/sewerina/meter_readings/database/ReadingEntity.kt
package com.github.sewerina.meter_readings.database

import androidx.room.*
import java.io.Serializable
import java.util.*

@Entity(
    tableName = "reading",
    foreignKeys = [ForeignKey(
        entity = HomeEntity::class,
        parentColumns = ["id"],
        childColumns = ["homeId"],
        onDelete = ForeignKey.CASCADE
    )],
    indices = [Index("homeId")]
)
class ReadingEntity(
    @JvmField
    @ColumnInfo(name = "homeId")
    var homeId: Int,

    @JvmField
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    var id: Int,

    @JvmField
    @ColumnInfo(name = "date")
    @TypeConverters(DateConverter::class)
    var date: Date,

    @JvmField
    @ColumnInfo(name = "coldWater")
    var coldWater: Int,

    @JvmField
    @ColumnInfo(name = "hotWater")
    var hotWater: Int,

    @JvmField
    @ColumnInfo(name = "drainWater")
    var drainWater: Int,

    @JvmField
    @ColumnInfo(name = "electricity")
    var electricity: Int,

    @JvmField
    @ColumnInfo(name = "gas")
    var gas: Int
) : Serializable

class NewReadingEntity(
    @JvmField
    @ColumnInfo(name = "date")
    @TypeConverters(DateConverter::class)
    var date: Date,

    @JvmField
    @ColumnInfo(name = "homeId")
    var homeId: Int = -1,

    @JvmField
    @ColumnInfo(name = "coldWater")
    var coldWater: Int = 0,

    @JvmField
    @ColumnInfo(name = "hotWater")
    var hotWater: Int = 0,

    @JvmField
    @ColumnInfo(name = "drainWater")
    var drainWater: Int = 0,

    @JvmField
    @ColumnInfo(name = "electricity")
    var electricity: Int = 0,

    @JvmField
    @ColumnInfo(name = "gas")
    var gas: Int = 0
) : Serializable {
    fun toEntity(id: Int): ReadingEntity {
        return ReadingEntity(homeId, id, date, coldWater, hotWater, drainWater, electricity, gas)
    }
}